tests/unittests/sources/test_ovf.py

```diff
@@ -192,52 +191,45 @@ def test_with_b64_network_config_disable_read_network(self):
         }
         env = fill_properties(props)
         md, ud, cfg = dsovf.read_ovf_environment(env)
-        self.assertEqual({"instance-id": "inst-001"}, md)
-        self.assertEqual({"password": "passw0rd"}, cfg)
-        self.assertIsNone(ud)
+        assert {"instance-id": "inst-001"} == md
+        assert {"password": "passw0rd"} == cfg
+        assert ud is None
 
 
-class TestDatasourceOVF(CiTestCase):
-    with_logs = True
+class TestDatasourceOVF:
+    @pytest.fixture
+    def ds(self, paths):
+        return dsovf.DataSourceOVF(sys_cfg={}, distro={}, paths=paths)
 
-    def setUp(self):
-        super(TestDatasourceOVF, self).setUp()
-        self.datasource = dsovf.DataSourceOVF
-        self.tdir = self.tmp_dir()
-
-    def test_get_data_seed_dir(self):
+    def test_get_data_seed_dir(self, ds, paths):
         """Platform info properly reports when getting data from seed dir."""
-        paths = Paths({"cloud_dir": self.tdir, "run_dir": self.tdir})
         # Write ovf-env.xml seed file
-        seed_dir = self.tmp_path("seed", dir=self.tdir)
-        ovf_env = self.tmp_path("ovf-env.xml", dir=seed_dir)
+        ovf_env = os.path.join(paths.seed_dir, "ovf-env.xml")
         util.write_file(ovf_env, OVF_ENV_CONTENT)
-        ds = self.datasource(sys_cfg={}, distro={}, paths=paths)
 
-        self.assertEqual("ovf", ds.cloud_name)
-        self.assertEqual("ovf", ds.platform_type)
+        assert "ovf" == ds.cloud_name
+        assert "ovf" == ds.platform_type
         with mock.patch(MPATH + "transport_vmware_guestinfo") as m_guestd:
             with mock.patch(MPATH + "transport_iso9660") as m_iso9660:
                 m_iso9660.return_value = NOT_FOUND
                 m_guestd.return_value = NOT_FOUND
-                self.assertTrue(ds.get_data())
-                self.assertEqual(
-                    "ovf (%s/seed/ovf-env.xml)" % self.tdir, ds.subplatform
+                assert ds.get_data()
+                assert (
+                    "ovf (%s/ovf-env.xml)" % paths.seed_dir == ds.subplatform
                 )
 
     @mock.patch("cloudinit.subp.subp")
     @mock.patch("cloudinit.sources.DataSource.persist_instance_data")
+    @pytest.mark.parametrize(
+        ["guestinfo", "iso"],
+        [
+            pytest.param(False, True, id="vmware_guestinfo"),
+            pytest.param(True, False, id="iso9660"),
+        ],
+    )
```

Comment:
It feels strange to have the parametrize declaration stacked lower than the mock.patches as the unittest params typically ordered left-to-right from lowest decorator in the stack. Let's put the parametrize up top.